packages/runfiles/runfiles.ts

```diff
@@ -116,6 +122,25 @@ export class Runfiles {
     return runfilesEntries;
   }
 
+  loadRepoMapping(runfilesDir: string): Map<string, string>|undefined {
+    const repoMappingPath = path.join(runfilesDir, REPO_MAPPING_RLOCATION);
+
+    if (fs.existsSync(repoMappingPath)) {
+      const repoMappings = new Map<string, string>()
+      const mappings = fs.readFileSync(repoMappingPath, {encoding: 'utf-8'});
+      for (const line of mappings.split('\n')) {
+        if (!line) continue;
+        const [from, repoName, repoPath] = line.split(',');
+
+        // TODO: from !== ''?
```

Comment:
TODO: do this https://github.com/bazelbuild/rules_go/pull/3347/files#diff-64181e823663efa1ae9c618e487f50c7de433e64d829ff2b59a93abf0989781aR53-R81